e2e/tests/workspace/workspace-actions-share.spec.ts

```diff
@@ -0,0 +1,119 @@
+import WorkspaceCard from 'app/component/workspace-card';
+import HomePage from 'app/page/home-page';
+import WorkspaceAboutPage from 'app/page/workspace-about-page';
+import WorkspacesPage from 'app/page/workspaces-page';
+import {LinkText, WorkspaceAccessLevel} from 'app/text-labels';
+import {config} from 'resources/workbench-config';
+import {findOrCreateWorkspace, signIn, signInAs, signOut} from 'utils/test-utils';
+import {makeWorkspaceName} from 'utils/str-utils';
+import WorkspaceDataPage from 'app/page/workspace-data-page';
+import {waitWhileLoading} from 'utils/waits-utils';
+
+
+describe('Share workspace', () => {
+
+  beforeEach(async () => {
+    await signIn(page);
+  });
+
+  // Assume there is at least one workspace preexist
+  describe('From the workspace about or data page via workspace actions menu', () => {
+
+    test('As OWNER, user can share a workspace', async () => {
+      
+      const workspaceCard = await findOrCreateWorkspace(page);
+      await workspaceCard.clickWorkspaceName();
+
+      await (new WorkspaceDataPage(page)).openAboutPage();
+      const aboutPage = new WorkspaceAboutPage(page);
+      await aboutPage.waitForLoad();
+
+      // This test is not hermetic - if the collaborator is already on this
+      // workspace, just remove them before continuing.
+      let accessLevel = await aboutPage.findUserInCollaboratorList(config.collaboratorUsername);
+      if (accessLevel !== null) {
+        await (await aboutPage.shareWorkspace()).removeUser(config.collaboratorUsername);
+        await waitWhileLoading(page);
+      }
```

Comment:
can you make this a func in about page? I think there are multiple files are using same code.

Reply:
will do. Thanks!